src/modules/esl-note/core/esl-note.ts

```diff
@@ -0,0 +1,175 @@
+import {ExportNs} from '../../esl-utils/environment/export-ns';
+import {attr, boolAttr, ESLBaseElement} from '../../esl-base-element/core';
+import {bind} from '../../esl-utils/decorators/bind';
+import {ready} from '../../esl-utils/decorators/ready';
+import {ESLTooltip} from '../../esl-tooltip/core';
+import {ESLFootnotes} from '../../esl-footnotes/core/esl-footnotes';
+import {EventUtils} from '../../esl-utils/dom/events';
+import {ENTER, SPACE} from '../../esl-utils/dom/keys';
+import {DeviceDetector} from '../../esl-utils/environment/device-detector';
+import {ESLMediaQuery} from '../../esl-media-query/core';
+
+import type {ToggleableActionParams} from '../../esl-toggleable/core/esl-toggleable';
+
+@ExportNs('Note')
+export class ESLNote extends ESLBaseElement {
+  static is = 'esl-note';
+  static eventNs = 'esl:note';
+
+  /** Linked state marker */
+  @boolAttr() public linked: boolean;
+
+  /** Tooltip state marker */
+  @boolAttr() public tooltipShown: boolean;
+
+  /** Click event tracking media query. Default: `all` */
+  @attr({defaultValue: 'all'}) public trackClick: string;
+  /** Hover event tracking media query. Default: `all` */
+  @attr({defaultValue: 'all'}) public trackHover: string;
+
+  @attr() public noteText: string;
+
+  protected _$footnotes: ESLFootnotes | null;
+  protected _index: number;
+  protected _innerHTML: string;
+
+  /** Marker to allow track hover */
+  public get allowHover() {
+    return DeviceDetector.hasHover && ESLMediaQuery.for(this.trackHover).matches;
+  }
+  /** Marker to allow track clicks */
+  public get allowClick() {
+    return ESLMediaQuery.for(this.trackClick).matches;
+  }
+
+  get index(): number {
+    return this._index;
+  }
+
+  set index(val: number) {
```

Comment:
Do we need it ? It is not safe for user in that moment

Reply:
removed